### PR TITLE
Finished documentation for gym

### PR DESCRIPTION
**Pull request by yuechen2001**
Finished my portion of the gym documentation for frontend  

- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in** the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
```
Comment (gunbux) [inline .vscode/launch.json:5]: Don't think this should be in the PR

File: frontend/src/components/Gym/GymKeyWith.tsx [hunk 2]
```diff
@@ -4,6 +4,21 @@ import Avatar from '../../assets/Avatar_4x.png'
...
+ * @returns user's profile picture, name and link to his/her telegram 
+ * 
+ * @example 
+ * user called Shaun and has no profile picture 
+ * component will display the default profile picture, his name Shaun as well as his 
+ * telegram handle. Clicking on the telegram handle redirects the user to key holder's telegram
+ * 
+ * @remarks 
+ * <any remarks on this component type here> 
+ *
```
Comment (gunbux) [inline frontend/src/components/Gym/GymKeyWith.tsx:19]: If there's no remarks, I think you can remove this

**Review by gunbux — DISMISSED** [review]
It looks like there are some merge conflicts as well, the automated tests in Github is failing as well, do make sure those are passing before we can merge it!

**Review by marcus-ong-qy — CHANGES_REQUESTED** [review]
apologies for the late review haha; do look at them accordingly and good job!

File: frontend/src/components/Gym/GymKeyWith.tsx [hunk 3]
```diff
@@ -52,6 +52,21 @@ interface Props {
...
+ * @param name (string)
+ * @param handle (string) user's telegram handle
+ * @param avatar (string, optional) user's profile picture. The string is of the form url
+ * @returns user's profile picture, name and link to his/her telegram
+ *
+ * @example
+ * user called Shaun and has no profile picture
+ * component will display the default profile picture, his name Shaun as well as his
+ * telegram handle. Clicking on the telegram handle redirects the user to key holder's telegram
+ * <GymKeyWith name={"thomaslim"} handle={"@thomas123"} avatar={""} />
```
Comment (marcus-ong-qy) [inline frontend/src/components/Gym/GymKeyWith.tsx:66]: btw JSX syntax string params don't need curly braces, just double quotes can alr
```suggestion
 * <GymKeyWith name="thomaslim" handle="@thomas123" avatar="" />
```

File: frontend/src/components/Gym/GymStatus.tsx [hunk 4]
```diff
@@ -33,6 +33,17 @@ interface Props {
...
 }
 
+/**
+ *
+ * @param isOpen (boolean) status of the gym
+ * @returns closed/open status of the gym
+ *
+ * @example
+ * Retrieves gym status from backend. If the gym is closed, display "Gym is Closed" in red
+ * <GymStatus isOpen={True} />
```
Comment (marcus-ong-qy) [inline frontend/src/components/Gym/GymStatus.tsx:43]: another JSX syntax nitpicking haha, boolean `={true}` is redundant
```suggestion
 * <GymStatus isOpen />
```

Comment (marcus-ong-qy) [reply frontend/src/components/Gym/GymStatus.tsx:43]: these are not wrong but just to standardise stylings around here abit; can refer to our style guides [here](https://www.notion.so/WIP-RHDevs-Frontend-Code-Style-Guide-b5f294c047f14f78b370031afb2c73c5) :)

File: frontend/src/components/Gym/GymStatusTab.tsx [hunk 5]
```diff
@@ -105,6 +154,14 @@ function GymStatusTab() {
     )
   }
 
+  /**
+   *
+   * @param
+   * @returns a button to be displayed on the screen, with onClick function handleModalState()
```
Comment (marcus-ong-qy) [inline frontend/src/components/Gym/GymStatusTab.tsx:160]:
```suggestion
   * @returns a button to be displayed on the screen, with onClick function handleModalState() \
```

**Review by marcus-ong-qy — CHANGES_REQUESTED** [review]
nice! just a few more comments

File: frontend/src/components/Gym/GymKeyWith.tsx [hunk 6]
```diff
@@ -52,6 +52,21 @@ interface Props {
...
+ * @param handle (string) user's telegram handle
+ * @param avatar (string, optional) user's profile picture. The string is of the form url
+ * @returns user's profile picture, name and link to his/her telegram
+ *
+ * @example
+ * user called Shaun and has no profile picture
+ * component will display the default profile picture, his name Shaun as well as his
+ * telegram handle. Clicking on the telegram handle redirects the user to key holder's telegram
+ * <GymKeyWith name="thomaslim" handle="@thomas123" avatar="" />
+ *
```
Comment (marcus-ong-qy) [inline frontend/src/components/Gym/GymKeyWith.tsx:67]:
```suggestion
 * @example
 * ```
 * // user called Shaun and has no profile picture
 * // component will display the default profile picture, his name Shaun as well as his
 * // telegram handle. Clicking on the telegram handle redirects the user to key holder's telegram
 * <GymKeyWith name="thomaslim" handle="@thomas123" avatar="" />
 * ```
 *
```

File: frontend/src/components/Gym/GymStatus.tsx [hunk 7]
```diff
@@ -33,6 +33,17 @@ interface Props {
...
 }
 
+/**
+ *
+ * @param isOpen (boolean) status of the gym
+ * @returns closed/open status of the gym
+ *
+ * @example
+ * Retrieves gym status from backend. If the gym is closed, display "Gym is Closed" in red
+ * <GymStatus isOpen />
```
Comment (marcus-ong-qy) [inline frontend/src/components/Gym/GymStatus.tsx:43]:
```suggestion
 * @example
 * ```
 * // Retrieves gym status from backend. If the gym is closed, display "Gym is Closed" in red
 * <GymStatus isOpen />
 * ```
```

**Review by marcus-ong-qy — APPROVED** [review]
make these changes then can merge alr!

**Pull request merged**